pkg/envoy/ciliumenvoyconfig.go

```diff
@@ -145,8 +145,14 @@ func ParseResources(cecNamespace string, cecName string, anySlice []cilium_v2.XD
 				listener.EnableReusePort = &wrapperspb.BoolValue{Value: false}
 			}
 
+			// Figure out if this is an internal listener
+			isInternalListener := listener.GetInternalListener() != nil
+
+			// Only inject Cilium filters if Cilium allocates listener address
+			injectCiliumFilters := listener.GetAddress() == nil && !isInternalListener
+
 			// Inject Cilium bpf metadata listener filter, if not already present.
-			{
+			if !isInternalListener {
```

Comment:
Perhaps we should emit a log here warning users that their envoy config may not have the outcome they expect?

Reply:
(I guess there's the bootup message but this would indicate that there is indeed such a listener)

Reply:
This is still a non-stable/experimental feature, so IMO logging is not (yet) necessary.